python/pyiceberg/expressions/literals.py

```diff
@@ -207,7 +207,7 @@ def __init__(self, value: int):
         super().__init__(value, int)
 
     @singledispatchmethod
-    def to(self, type_var: IcebergType) -> Literal:
+    def to(self, type_var: IcebergType) -> Literal:  # type: ignore
```

Comment:
Is there no way to return an unknown type from a method in Python? In Java, we would just declare that the method returns `Literal<T>` where `T` comes from context. Then you get a type that may not be correct, but is at least not `<nothing>`.

Reply:
I don't like this either, the problem is that it is unbound. I have some idea's on how to fix this properly in https://github.com/apache/iceberg/issues/6231